BennyTordrupVisma/Week2/Ass3/script.js

```diff
@@ -0,0 +1,340 @@
+var Message = React.createClass({
+	getInitialState: function() {
+		return {
+			listSelectionVisible : false
+		}
+	},
+	
+	render: function() {
+		return (
+			<div>
+					{this.props.message.isArchived 
+						? <div>
+							<div className="archived-message">
+								{this.props.message.id}. {this.props.message.text}
+								<button onClick={this.handleUnarchive}>Unarchive</button>
+							</div>	
+						  </div>
+						: <div>
+							<div>
+								{this.props.message.id}. {this.props.message.text}
+								<button onClick={this.handleArchive}>Archive</button>
+								<button onClick={this.handleDelete}>Delete</button>
+								{!this.state.listSelectionVisible 
+									? <button onClick={this.handleShowListSelection}>Move</button> 
+									: null}
+								{this.state.listSelectionVisible 
+									? <div>
+										<label>New list: </label>
+										<input type="text" ref="newList" />
+										<button onClick={this.handleMoveMessage}>Move</button>
+									  </div> 
+									: null}
```

Comment:
You can use the `&&` syntax which will evaluate the second expression only if the first expression evaluates to something else than `undefined, null, '', 0`. 

```jsx
{this.state.listSelectionVisible &&
  <div>
    <label>New list: </label>
    <input type="text" ref="newList" />
    <button onClick={this.handleMoveMessage}>Move</button>
  </div>}
```